lookout/style/format/benchmarks/generate_smoke.py

```diff
@@ -0,0 +1,115 @@
+    "no_newline_before_function": ("(\n+)function ", "\nfunction "),
+    "two_newlines_before_function": ("(\n+)function ", "\n\nfunction "),
+    "no_space_after_if": (r"if\s*\(", r"if("),
+    "no_spaces_inside_brackets": (r"\{ +([^{}]+) +\}", r"{\1}"),
+    "no_spaces_inside_round_brackets": (r"\( +([^()]+) +\)", r"(\1)"),
+    "spaces_inside_round_brackets": (r"\(([^()]+)\)", r"( \1 )"),
+}
+
+
+def commit_style(repo: Repo, format_rule_name: str) -> Tuple[str, str]:
+    """
+    Call bash script which commit all changes to `style_name` branch and checkout master back.
+
+    :param repo: Repo instance to the repository for which style were applied.
+    :param format_rule_name: Applied format rule name.
+    :return: Two commit hashes: where style was applied and where style was disrupt.
+    """
+    repopath = repo.path
+    base = repo.head()
+    # Checkout new branch
+    branch_create(repopath, format_rule_name, force=True)
+    update_head(repopath, format_rule_name)
+    # Commit everything
+    for tree_path, entry in repo.open_index().iteritems():
+        full_path = os.path.join(repopath.encode(), tree_path)
+        blob = blob_from_path_and_stat(full_path, os.lstat(full_path))
+        if blob.id != entry.sha:
+            repo.stage(tree_path)
+    style_commit = repo.do_commit(format_rule_name.encode(),
+                                  b"Source{d} ML Team <ml@sourced.tech>")
+    # Back to master
+    update_head(repopath, b"master")
+    build_index_from_tree(repo.path, repo.index_path(), repo.object_store, repo[base].tree)
+
+    return style_commit.decode(), base.decode()
+
+
+def generate_smoke_entry(inputpath: str, outputpath: str, force: bool = False) -> int:
+    """
+    Generate repositories with different style in separate branches and its violations.
+
+    :param inputpath: Path to the tar.gz archive containing initial repositories.
+    :param outputpath: Path to the directory where the generated dataset should be stored.
+    :param force: Override outputpath directory if exists.
+
+    :return: Success status. 0 if Succeeded, 1 otherwise.
+    """
+    log = logging.getLogger("styles-gen")
+    if not inputpath.endswith(".tar.xz"):
+        raise ValueError("Input file should be .tar.xz archive.")
+    inputpath = Path(inputpath)
+    outputpath = Path(outputpath)
+    if force and outputpath.exists():
+        shutil.rmtree(str(outputpath))
+    try:
+        outputpath.mkdir()
+    except FileExistsError:
+        log.error("Directory %s exists. If you want to override it run with --force flag.",
+                  outputpath)
+        return 1
+    tarfile.open(str(inputpath)).extractall(str(outputpath))
+    repopaths = [x for x in outputpath.iterdir() if x.is_dir()]
+    log.info("Repos found: %s", ", ".join(r.stem for r in repopaths))
+    with open(str(outputpath / index_filename), "w") as index_file:
+        index_file.write("repo,style,from,to\n")
```

Comment:
this is why we have [`csv.writer`](https://docs.python.org/3/library/csv.html#csv.writer) in the stdlib